### PR TITLE
grt: bug fix for repair_antennas

### PR DESCRIPTION
**Pull request by eder-matheus**

### CODE REVIEW

File: src/grt/src/GlobalRouter.cpp [hunk 1]
```diff
@@ -1585,7 +1585,7 @@ void GlobalRouter::addGuidesForPinAccess(odb::dbNet* db_net, GRoute& route)
                       || route[i].final_layer < wire_via_layer))) {
             // remove all vias to this pin that doesn't connects two wires
             route.erase(route.begin() + i);
-            i = 0;
+            i--;
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:1588]: Is this just a runtime optimization?

File: src/grt/src/GlobalRouter.cpp [hunk 2]
```diff
@@ -1903,23 +1903,43 @@ void GlobalRouter::addLocalConnections(NetRouteMap& routes)
...
-
-      route.push_back(hor_segment);
-      route.push_back(ver_segment);
+
+      bool segment_overlaps_pin = false;
+      for (const odb::Rect& box : pin_boxes) {
+        if ((segment_overlaps_pin = box.overlaps(pin_position))) {
+          break;
+        }
+      }
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:1912]: if segment_overlaps_pin can we just skip the rest (i.e. continue)?

File: src/grt/src/GlobalRouter.cpp [hunk 3]
```diff
@@ -1903,23 +1903,43 @@ void GlobalRouter::addLocalConnections(NetRouteMap& routes)
...
+
+      int minor_distance = std::numeric_limits<int>::max();
```
Comment (maliberty) [inline src/grt/src/GlobalRouter.cpp:1914]: minimum_distance rather than minor_distance

**Review by maliberty — CHANGES_REQUESTED** [review]

File: src/grt/src/GlobalRouter.cpp (hunk 1 above)
Comment (eder-matheus) [reply src/grt/src/GlobalRouter.cpp:1588]: Not really. It also makes sure all segments in the vector are evaluated after a via is deleted.

**Review by maliberty — APPROVED** [review]

**Pull request merged**